webhook_server/libs/graphql/unified_api.py

```
@@ -0,0 +1,1779 @@
+                TransportConnectionFailed,
+            ) as ex:
+                # Fallback to REST API if GraphQL fails (including transport errors)
+                self.logger.warning(f"GraphQL failed to get commits, falling back to REST: {ex}")
+                # Get REST PullRequest from wrapper's rest_pr attribute
+                rest_commits = await self.get_pr_commits_rest(
+                    actual_pull_request.rest_pr
+                    if actual_pull_request and hasattr(actual_pull_request, "rest_pr")
+                    else None
+                )
+                if not rest_commits:
+                    raise ValueError(f"No commits found in PR {actual_pr_number}") from ex
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)